pkg/proxy/endpoints.go

```diff
@@ -104,7 +105,8 @@ type EndpointChangeTracker struct {
 	// items maps a service to is endpointsChange.
 	items map[types.NamespacedName]*endpointsChange
 	// makeEndpointInfo allows proxier to inject customized information when processing endpoint.
-	makeEndpointInfo makeEndpointFunc
+	makeEndpointInfo   makeEndpointFunc
+	endpointsMapChange endpointsMapChangeFunc
```

Comment:
Maybe just a bit of a rename here to make it clear this is a function and not a change object.
```suggestion
	onEndpointsMapChange endpointsMapChangeFunc
```